packages/aws-cdk-lib/aws-rds/lib/validate-database-cluster-props.ts

```diff
@@ -0,0 +1,54 @@
+import { Construct } from 'constructs';
+import { ClusterScailabilityType, DatabaseCluster, DatabaseClusterProps, DBClusterStorageType } from './cluster';
+import { PerformanceInsightRetention } from './props';
+import { validateProps, ValidationRule } from '../../core/lib/helpers-internal';
+
+const standardDatabaseRules: ValidationRule<DatabaseClusterProps>[] = [
+  {
+    condition: (props) => props.enablePerformanceInsights === false &&
+        (props.performanceInsightRetention !== undefined || props.performanceInsightEncryptionKey !== undefined),
+    message: () => '`enablePerformanceInsights` disabled, but `performanceInsightRetention` or `performanceInsightEncryptionKey` was set',
+
+  },
+];
+
+const limitlessDatabaseRules: ValidationRule<DatabaseClusterProps>[] = [
+  {
+    condition: (props) => !props.enablePerformanceInsights,
+    message: () => 'Performance Insights must be enabled for Aurora Limitless Database',
+  },
+  {
+    condition: (props) => !props.performanceInsightRetention
+      || props.performanceInsightRetention < PerformanceInsightRetention.MONTHS_1,
+    message: () => 'Performance Insights retention period must be set to at least 31 days for Aurora Limitless Database',
+  },
+  {
+    condition: (props) => !props.monitoringInterval || !props.enableClusterLevelEnhancedMonitoring,
+    message: () => 'Cluster level enhanced monitoring must be set for Aurora Limitless Database. Please set \'monitoringInterval\' and enable \'enableClusterLevelEnhancedMonitoring\'',
+  },
+  {
+    condition: (props) => !!(props.writer || props.readers),
+    message: () => 'Aurora Limitless Database does not support reader or writer instances',
+  },
+  {
+    condition: (props) => !props.engine.engineVersion?.fullVersion?.endsWith('limitless'),
+    message: (props) => `Aurora Limitless Database requires an engine version that supports it, got: ${props.engine.engineVersion?.fullVersion}`,
+  },
+  {
+    condition: (props) => props.storageType !== DBClusterStorageType.AURORA_IOPT1,
+    message: (props) => `Aurora Limitless Database requires I/O optimized storage type, got: ${props.storageType}`,
+  },
+  {
+    condition: (props) => props.cloudwatchLogsExports === undefined || props.cloudwatchLogsExports.length === 0,
+    message: () => 'Aurora Limitless Database requires CloudWatch Logs exports to be set',
+  },
+];
+
+export function validateDatabaseClusterProps(scope: Construct, props: DatabaseClusterProps): void {
+  const isLimitlessCluster = props.clusterScailabilityType === ClusterScailabilityType.LIMITLESS;
+  const applicableRules = isLimitlessCluster
+    ? [...standardDatabaseRules, ...limitlessDatabaseRules]
+    : standardDatabaseRules;
+
+  validateProps(scope, DatabaseCluster.name, props, applicableRules);
```

Comment:
Is using `DatabaseCluster.name` always the correct choice?

If someone implements a subclass of `DatabaseCluster` named `FooBar`, the error will still say "DatabaseCluster". Is that what we want? Genuine question, cause I'm not sure about it.

Reply:
I did not consider that while implementing this, but I think that is the correct behavior. Ultimately we are validating the props of `DatabaseCluster`, not `FooBar`. If there are additional validations needed for `FooBar`, they should be implemented separately.